### PR TITLE
Made Article schema. Made create and get article by id and by topic

### PR DESCRIPTION
**Pull request by HarisShahidHameed**
Trello Link : https://trello.com/c/GZH1usu6/22-create-article-blog-schema

### CODE REVIEW

File: server/DataBase/models/article.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const mongoose = require('mongoose');
+const uuidv4 = require('uuid/v4');
+
+
```
Comment (abdulhaqshah) [inline server/DataBase/models/article.js]: one line space only

File: server/libs/articlelibs.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
+const Article = require ('./../DataBase/models/article');
+const {ObjectID} = require('mongodb');
+const {statusCodes, messages, secretKeys, timeScale} = require ('../utilities/constants');
+
+const createArticle = function (data) {
+    return new Promise((resolve,reject) => {
+        Article.create({
+            publishedBy : data.uuid,
+            topic : data.body.topic,
```
Comment (abdulhaqshah) [inline server/libs/articlelibs.js]: JUST DATA.TOPIC NO BODY HERE

File: server/libs/articlelibs.js [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+            article : data.body.article
```
Comment (abdulhaqshah) [inline server/libs/articlelibs.js]: same as above

File: server/routes/article.js [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        uuid : req.header('uuid')
+    }
+    articleLibs.createArticle(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+});
+
+router.get('/article/articleById/:articleId', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: /article/:articleId

File: server/routes/article.js [hunk 5]
```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+const router = express.Router();
+const articleLibs = require('../libs/articlelibs');
+const {authenticate} = require('../middleware/authenticate');
+
+router.post('/article/newArticle', (req, res) => {
+    let data = {
+        body : req.body,
```
Comment (abdulhaqshah) [inline server/routes/article.js]: fields not body extract value from the body or you can do like data = req.body

File: server/routes/article.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+const router = express.Router();
+const articleLibs = require('../libs/articlelibs');
+const {authenticate} = require('../middleware/authenticate');
+
+router.post('/article/newArticle', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: /article/

File: server/routes/article.js [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+        articleId : req.params.articleId
+    }
+    articleLibs.getArticle(query).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+});
+
+router.get('/article/articleByTopic', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: /article/:topic

File: server/routes/article.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+    let query = {
+        topic : req.body.topic
```
Comment (abdulhaqshah) [inline server/routes/article.js]: value should come in params not body for getting

File: server/routes/article.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    articleLibs.getArticle(query).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+});
+
+router.patch('/article/addComment', (req, res) => {
+    let data = {
+        body : req.body,
```
Comment (abdulhaqshah) [inline server/routes/article.js]: fields not body extract value from the body or you can do like data = req.body

File: server/routes/article.js [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+router.patch('/article/editComment', (req, res) => {
+    let data = req.body;
+    articleLibs.editComment(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+})
+
+router.get('/article/getComment', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: getting all comments should be article/comments. Its getting all the comments in the app or related to specific article if specific article is should have articleId in params

File: server/routes/article.js [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    let data = req.body;
```
Comment (abdulhaqshah) [inline server/routes/article.js]: req.body is doing in getting comments

File: server/routes/article.js [hunk 12]
```diff
@@ -0,0 +1,79 @@
+const express = require('express');
+const router = express.Router();
+const articleLibs = require('../libs/articlelibs');
+const {authenticate} = require('../middleware/authenticate');
+
```
Comment (abdulhaqshah) [inline server/routes/article.js:5]: where is the route of getAll articles

File: server/routes/article.js [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+router.get('/article/getComment', (req, res) => {
+    let data = req.body;
+    articleLibs.getCommentById(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+})
+
+router.delete('/article/deleteComment', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: same as above if its related to specific article its id should b in params and the commented deleted Id too.

**Review by abdulhaqshah — CHANGES_REQUESTED** [review]

File: server/DataBase/models/article.js (hunk 1 above)
Comment (HarisShahidHameed) [reply server/DataBase/models/article.js]: Removed

File: server/libs/articlelibs.js (hunk 2 above)
Comment (HarisShahidHameed) [reply server/libs/articlelibs.js]: Done

File: server/libs/articlelibs.js (hunk 3 above)
Comment (HarisShahidHameed) [reply server/libs/articlelibs.js]: Done

File: server/routes/article.js (hunk 4 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 5 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 6 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 7 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 8 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 9 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 10 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 11 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Removed.

File: server/routes/article.js (hunk 12 above)
Comment (HarisShahidHameed) [reply server/routes/article.js:5]: Done

File: server/routes/article.js (hunk 13 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: .eslintrc.json [hunk 14]
```diff
@@ -70,7 +70,7 @@
             }
         ],
         "eol-last": "off",
-        "eqeqeq": "error",
+        "eqeqeq": "off",
```
Comment (abdulhaqshah) [inline .eslintrc.json]: why off?

File: server/libs/articlelibs.js [hunk 15]
```diff
@@ -0,0 +1,159 @@
...
+            comment : data.comment,
+            commentedBy : data.uuid
+        }
+        Article.findOneAndUpdate({articleId : data.articleId}, {$push : {comments : comment}}, {new : true}).then((article) => {
+            if (article) {
+                let coment = article.comments.find((comments) => comments.comment === data.comment);
+                resolve({
+                    status : statusCodes.successful,
+                    message : `Comment ${messages.added}`,
+                    data : coment._id
```
Comment (abdulhaqshah) [inline server/libs/articlelibs.js]: why sending _id in data

File: server/routes/article.js [hunk 16]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    articleLibs.getArticle(query).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+});
+
+router.get('/article/:topic', (req, res) => {
+    let query = {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: not need to create a variable for single field in object just do articleLibs.getArticle({topic : req.params.topic})

File: server/routes/article.js [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+    }
+    articleLibs.addComment(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+})
+
+router.patch('/article/editComment', (req, res) => {
+    let data = req.body;
```
Comment (abdulhaqshah) [inline server/routes/article.js]: no need for this variable just pass directly req.body in the function

File: server/routes/article.js [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+    articleLibs.editComment(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+})
+
+router.get('/article/getCommentById/:aid/:cid', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: /article/:aid/comment/:cid

File: server/routes/article.js [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+        commentId : req.params.cid
+    }
+    articleLibs.getCommentById(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+})
+
+router.delete('/article/deleteComment/:aid/:cid', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: /article/:aid/comment/:cid/delete

File: server/routes/article.js [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+        commentId : req.params.cid
+    }
+    articleLibs.deleteComment(data).then((article) => {
+        res.status(article.status).send(article);
+    }).catch((error) => {
+        res.status(error.status).send(error);
+    });
+})
+
+router.get('/article/getAllArticles', (req, res) => {
```
Comment (abdulhaqshah) [inline server/routes/article.js]: /article/ or /article/all

File: server/routes/article.js [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+    let query = {};
```
Comment (abdulhaqshah) [inline server/routes/article.js]: no variable needed just pass empty object in the function

File: tests/article.test.js [hunk 22]
```diff
@@ -0,0 +1,108 @@
...
+const {ObjectID} = require('mongodb');
+const defaults = require('superagent-defaults');
+const {app} = require('../server/app');
+const uuidv4 = require('uuid/v4');
+const Article = require('../server/DataBase/models/article');
+const {articles, populateArticles} = require('./articleseed');
+const {secretKeys} = require ('./../server/utilities/constants');
+const {users, populateUsers} = require('./seed');
+
+let articleOne;
```
Comment (abdulhaqshah) [inline tests/article.test.js]: you can write this in one like let articleOne, articleTwo and so on

**Review by abdulhaqshah — CHANGES_REQUESTED** [review]

File: .eslintrc.json (hunk 14 above)
Comment (HarisShahidHameed) [reply .eslintrc.json]: Done

File: server/libs/articlelibs.js (hunk 15 above)
Comment (HarisShahidHameed) [reply server/libs/articlelibs.js]: Sending full comment now.

File: server/routes/article.js (hunk 16 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 17 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 18 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 19 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 20 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: server/routes/article.js (hunk 21 above)
Comment (HarisShahidHameed) [reply server/routes/article.js]: Done

File: tests/article.test.js (hunk 22 above)
Comment (HarisShahidHameed) [reply tests/article.test.js]: Done

**Review by abdulhaqshah — APPROVED** [review]

**Pull request merged**